wordpress-mobile/WordPress-Android
# [Jetpack] Configure Sentry

## Context

This PR configures a separate Sentry DSN for Jetpack.

The Sentry DSN for WordPress is defined in the `gradle.properties` file via `wp.sentry.dsn`, and those `wp.*`-prefixed properties are converted to `buildConfigField` in `build.gradle` and ends up being used as `BuildConfig.SENTRY_DSN` in the code.
This PR thus makes it possible for us to also define `jp.*`-prefixed properties, especially `jp.sentry.dsn`, to override the `buildConfigField` for the Jetpack app flavor.

The PR also duplicates the `sentry.properties` file (mobile-secrets file) which was at the repo root to instead use a different one for WP and JP. This file is used by the sentry gradle plugin to upload the Proguard Mappings file to the correct project in Sentry, thus needing different files for the 2 flavors here too. (Draft post to come to explain this – paqN3M-lr-p2)

## To Test

### Pull the latest secrets

 - Remove the old `rm sentry.properties` file
 - Run `./gradlew configureApply` to pull the latest `gradle.properties` file (containing the new `jp.sentry.dsn` key) and the 2 new `sentry.properties` separate files for WP and JP.
 - Check the new files are present in `./WordPress/src/{wordpress,jetpack}/sentry.properties`.

### Verify generated `BuildConfig` constants

 - Remove previously build-generated files: `rm -rf WordPress/build/generated/`
 - [WP] Run `./gradlew -q WordPress:generateWordpressVanillaReleaseBuildConfig`
 - [WP] `grep SENTRY_DSN WordPress/build/generated/source/buildConfig/wordpressVanilla/release/org/wordpress/android/BuildConfig.java` and check that the Java constant contains the same value as our `wp.sentry.dsn` property
 - [JP] Run `./gradlew -q WordPress:generateJetpackVanillaReleaseBuildConfig`
 - [JP] `grep SENTRY_DSN WordPress/build/generated/source/buildConfig/jetpackVanilla/release/org/wordpress/android/BuildConfig.java` and check that the Java constant contains the same value as our `jp.sentry.dsn` property

### Check Proguard Mapping files + Generate a fake crash

 - Build the `jetpackVanillaRelease` variant from Android Studio and install it
 - Look at the UUID that was printed after the `Newly uploaded debug symbols:` log at the very end of the build log (just before `BUILD SUCCESSFUL`)
 - Open [the Proguard tab of Sentry's jetpack-android project](https://sentry.io/settings/a8c/projects/jetpack-android/proguard/) and check that the mapping with the same UUID has been uploaded there.
 - Use [this secret feature](https://github.com/wordpress-mobile/WordPress-Android/pull/12707) (Me -> App Settings -> Jetpack for Android -> tap 10+ times below the "automattic.com") to force a dummy crash in the app
 - Go check in [the `jetpack-android` Sentry project](https://sentry.io/organizations/a8c/projects/jetpack-android/?project=5731682) that a new event of the crash appears there ([example here](https://sentry.io/organizations/a8c/issues/2497189760/?project=5731682))
 - Repeat the above steps with the WordPress app, to ensure that WP crashes are still reported in the `wordpress-android` Sentry project (and not the `jetpack-android` project)
